.circleci/config.yml

```diff
@@ -0,0 +1,64 @@
+version: 2.1
+
+commands:
+  yarn-install:
+    steps:
+      - restore_cache:
+          name: Restore Yarn Cache
+          keys:
+            - yarn-{{ checksum "yarn.lock" }}
+      - run:
+          name: Yarn Install
+          command: yarn install --frozen-lockfile
+      - save_cache:
+          name: Save Yarn Cache
+          key: yarn-{{ checksum "yarn.lock" }}
+          paths:
+            - ~/.cache/yarn
+
+jobs:
+  checks:
+    parameters:
+      platform:
+        type: string
+        default: ""
+      check-tests:
+        type: boolean
+        default: false
+      check-correctness:
+        type: boolean
+        default: false
+    docker:
+      - image: circleci/node:8
+    steps:
+      - checkout
+      - run:
+          name: Checkout Gutenberg
+          command: |
+            git submodule init
+            git submodule update
```

Comment:
We are using the `--recursive` option to git submodule update in Travis so far and, I think it'd be better to replicate that here too. It would also give us the flexibility to add other submodules at any point the future without having to remember to come and change the CI configuration.

In case this here is taken from [the README page](https://github.com/wordpress-mobile/gutenberg-mobile#clone-the-project), I think we should probably update the README to include the `--recursive` too (but not necessarily in this PR).

Reply:
Updated now 😄